java/org/apache/catalina/util/SessionConfig.java

```diff
@@ -49,38 +42,30 @@ public static String getSessionCookieName(Context context) {
      * @return the parameter name for the session
      */
     public static String getSessionUriParamName(Context context) {
-
-        String result = getConfiguredSessionCookieName(context);
-
-        if (result == null) {
-            result = DEFAULT_SESSION_PARAMETER_NAME;
-        }
-
-        return result;
+        return getConfiguredSessionCookieName(context, DEFAULT_SESSION_PARAMETER_NAME);
     }
 
 
-    private static String getConfiguredSessionCookieName(Context context) {
-
+    private static String getConfiguredSessionCookieName(Context context, String defaultName) {
+        if (context == null) {
+            return defaultName;
+        }
         // Priority is:
         // 1. Cookie name defined in context
         // 2. Cookie name configured for app
         // 3. Default defined by spec
-        if (context != null) {
-            String cookieName = context.getSessionCookieName();
```

Comment:
The code will be cleaner if you keep the original null check. That will also keep the code better aligned with the documented priority order.

Reply:
Thank you for your attention to detail in the code. 
Could you please check the modified code?

Reply:
LGTM